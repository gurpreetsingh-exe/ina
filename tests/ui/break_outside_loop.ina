// COMPILE_FAIL

fn main() {
    let a = break;
    if true {
        break
    }
    match true {
        _ => break,
    }
    { break }
    // OK
    loop { break; }
}
