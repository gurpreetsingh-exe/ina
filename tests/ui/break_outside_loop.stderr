error: break outside of a loop
   ╭─[tests/ui/break_outside_loop.ina:4:13]
   │
 4 │     let a = break;
   │             ──┬──
   │               ╰─── cannot `break` outside of a loop
   ∙
───╯
error: break outside of a loop
   ╭─[tests/ui/break_outside_loop.ina:6:9]
   │
 6 │         break
   │         ──┬──
   │           ╰─── cannot `break` outside of a loop
   ∙
───╯
error: break outside of a loop
   ╭─[tests/ui/break_outside_loop.ina:9:14]
   │
 9 │         _ => break,
   │              ──┬──
   │                ╰─── cannot `break` outside of a loop
   ∙
───╯
error: break outside of a loop
    ╭─[tests/ui/break_outside_loop.ina:11:7]
    │
 11 │     { break }
    │       ──┬──
    │         ╰─── cannot `break` outside of a loop
    ∙
────╯
